internal/worker/bootstrap/worker.go

```diff
@@ -403,7 +389,14 @@ func (w *bootstrapWorker) scopedContext() (context.Context, context.CancelFunc)
 }
 
 func (w *bootstrapWorker) seedAgentBinary(ctx context.Context, dataDir string) (func(), error) {
-	objectStore, err := w.cfg.ObjectStoreGetter.GetObjectStore(ctx, w.cfg.SystemState.ControllerModelUUID())
+	controllerModel, err := w.cfg.ModelService.ControllerModel(ctx)
```

Comment:
Could get this at instantiation. The data never changes.

Reply:
Done & Done